src/gc.c

```diff
@@ -77,7 +77,15 @@ static void threadCallback(void* data) {
   }
 
   if (!ret) {
-    rm_free (task);
+    // The index was freed. There is no need to reschedule the task.
+    // We need to free the task and the GC.
+    RedisModule_Log(RSDummyContext, REDISMODULE_LOGLEVEL_DEBUG, "GC: Self-Terminating. Index was freed.");
+    gc->callbacks.onTerm(gc->gcCtx);
+    if (bc) { // A debug command is waiting for the GC to finish
```

Comment:
I wonder why we need the `task->debug` that we check above.. here we do not check it, maybe we do not need it and we can simply count on `bc` not set to NULL?

Reply:
there is a non-blocking debug command for triggering the GC. we do need to know if this task is debug or not